renderer/canvas.py

```diff
@@ -221,6 +231,154 @@ def _draw_container_image(self, surface, container_name, obs, position):
+
+        Args:
+            tiling (String): Path to the tiling asset folder
+        """
+        # Load floor config
+        tiling_config_path = "tileset/" + tiling + "/config.json" # Assumes the name of the json is standardized as config.json
+        with open(os.path.join(RobotouilleCanvas.ASSETS_DIRECTORY, tiling_config_path), "r") as f:
+            tiling_config = json.load(f)
+
+        # Load and slice flooring spritesheet
+        spritesheet_path = "tileset/" + tiling + "/" + tiling_config["asset"]
+        spritesheet = pygame.image.load(os.path.join(RobotouilleCanvas.ASSETS_DIRECTORY, spritesheet_path)).convert_alpha()
+        num_sprites_x = tiling_config["columns"]
+        num_sprites_y = tiling_config["rows"]
+        sprite_width = spritesheet.get_width() // num_sprites_x
+        sprite_height = spritesheet.get_height() // num_sprites_y
+        sprites = []
+        for y in range(num_sprites_y):
+            for x in range(num_sprites_x):
+                rect = pygame.Rect(x * sprite_width, y * sprite_height, sprite_width, sprite_height)
+                sprite = spritesheet.subsurface(rect).convert_alpha()
+                sprites.append(sprite)
+        # Add empty sprite for index -1
+        empty = pygame.Surface((1, 1), flags=pygame.SRCALPHA)
+        empty.fill((0, 0, 0, 0))
+        sprites.append(empty)
+                
+        # Parse tile ID mappings
+        mappings = {}
+        for word, walls in tiling_config["mappings"].items():
+            entry = {}
+            for data in walls:
+                wall = data[0]
+                corners = {}
+                for t in data[1]:
+                    corners[t[0]] = t[1]
+                entry[wall] = corners
+            mappings[word] = entry
+
+        # Store loaded values in catalog
+        catalog = {}
+        catalog["config"] = tiling_config
+        catalog["sprites"] = sprites
+        catalog["mappings"] = mappings
+        return catalog
+        #self.asset_directory[tiling] = catalog
+    
+    def _parse_abstract_tile_matrix(self, abstract_matrix, tiling_catalog):
+        """
+        Parse the given abstract tile matrix into a matrix of raw tile IDs suitable for drawing
+
+        Args:
+            abstract_matrix (List[String]): List of strings whose characters represent abstract tiles
+            tiling_catalog (Dict): Catalog to reference when translating tilings
+        """
+        raw_matrix = [[-1] * len(abstract_matrix[0]) for _ in range(len(abstract_matrix))]
+        mappings = tiling_catalog["mappings"]
+
+        for row in range(len(abstract_matrix)):
+            for column in range(len(abstract_matrix[row])):
+                letter = abstract_matrix[row][column]
+                if not letter in mappings:
+                    raw_matrix[row][column] = -1
+                    continue
+
+                N_wall = "1" if row == 0 or abstract_matrix[row - 1][column] != letter else "0"
```

Comment:
comments would make this easier to skim through
`# Add wall if at top row or a foreign tile is above this tile`